src/main/java/com/wultra/security/userdatastore/model/repository/DocumentRepository.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * User Data Store
+ * Copyright (C) 2024 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.security.userdatastore.model.repository;
+
+import com.wultra.security.userdatastore.model.entity.DocumentEntity;
+import com.wultra.security.userdatastore.model.entity.UserClaimsEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+/**
+ * Repository for {@link DocumentEntity}.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ */
+@Repository
+public interface DocumentRepository extends JpaRepository<DocumentEntity, String> {
+
+    List<DocumentEntity> findAllByUserId(String userId);
+
+    int deleteAllByUserId(String userId);
```

Comment:
Since the last bad experience with performance, don't we want to delete using JPQL?

Reply:
We do not need to optimize this too much, there will be only several documents, photos or attachments per user.